operator/controllers/kyma_controller.go

```diff
@@ -152,7 +152,9 @@ func (r *KymaReconciler) synchronizeRemote(ctx context.Context, kyma *v1alpha1.K
 	if err := syncContext.SynchronizeRemoteKyma(ctx, remoteKyma); err != nil {
 		return fmt.Errorf("could not synchronize remote kyma: %w", err)
 	}
-
+	if err := syncContext.InsertWatcherLabels(ctx, remoteKyma); err != nil {
```

Comment:
The label can be inserted during remote Kyma creation in `CreateOrFetchRemoteKyma()`.
Also you do not need an explicit `Update()` call here, please combine it with`SynchronizeRemoteKyma()`, since `LastSync` would then not be valid.

Reply:
It will be moved to `SynchronizeRemoteKyma ` to cover also scenarios, in which the user accidentally removed the labels. Thus, we have covered all scenarios